spring-cloud-stream-binder-kafka-core/src/test/java/org/springframework/cloud/stream/binder/kafka/properties/KafkaBinderConfigurationPropertiesTest.java

```diff
@@ -105,4 +106,38 @@ public void mergedConsumerConfigurationFiltersEnableAutoCommitFromKafkaBinderCon
 
 		assertThat(mergedConsumerConfiguration).doesNotContainKeys(ConsumerConfig.GROUP_ID_CONFIG);
 	}
+
+	@Test
+	public void testCertificateFilesAreConvertedToAbsolutePathsFromClassPathResources() {
+		KafkaProperties kafkaProperties = new KafkaProperties();
+		KafkaBinderConfigurationProperties kafkaBinderConfigurationProperties =
+				new KafkaBinderConfigurationProperties(kafkaProperties);
+		final Map<String, String> configuration = kafkaBinderConfigurationProperties.getConfiguration();
+		configuration.put("ssl.truststore.location", "classpath:testclient.truststore");
+		configuration.put("ssl.keystore.location", "classpath:testclient.keystore");
+
+		kafkaBinderConfigurationProperties.getKafkaConnectionString();
+
+		assertThat(configuration.get("ssl.truststore.location"))
+				.isEqualTo(Paths.get(System.getProperty("java.io.tmpdir"), "testclient.truststore").toString());
+		assertThat(configuration.get("ssl.keystore.location"))
+				.isEqualTo(Paths.get(System.getProperty("java.io.tmpdir"), "testclient.keystore").toString());
+	}
+
+	@Test
+	public void testCertificateFilesAreConvertedToGivenAbsolutePathsFromClassPathResources() {
+		KafkaProperties kafkaProperties = new KafkaProperties();
+		KafkaBinderConfigurationProperties kafkaBinderConfigurationProperties =
+				new KafkaBinderConfigurationProperties(kafkaProperties);
+		final Map<String, String> configuration = kafkaBinderConfigurationProperties.getConfiguration();
+		configuration.put("ssl.truststore.location", "classpath:testclient.truststore");
+		configuration.put("ssl.keystore.location", "classpath:testclient.keystore");
+		kafkaBinderConfigurationProperties.setTruststoreLocationOnFileSystem("/var/tmp/foo.truststore");
+		kafkaBinderConfigurationProperties.setKeystoreLocationOnFileSystem("/var/tmp/bar.keystore");
```

Comment:
Will this test pass on Windows?

I don't remember if JUnit4 can be conditional on OS, I know JUnit5 can.